synapse/handlers/e2e_keys.py

```diff
@@ -259,34 +292,60 @@ def claim_client_keys(destination):
             ),
         )
 
+        opentracing.log_kv({"one_time_keys": json_result, "failures": failures})
         return {"one_time_keys": json_result, "failures": failures}
 
     @defer.inlineCallbacks
+    @opentracing.tag_args
     def upload_keys_for_user(self, user_id, device_id, keys):
+
         time_now = self.clock.time_msec()
 
         # TODO: Validate the JSON to make sure it has the right keys.
         device_keys = keys.get("device_keys", None)
+        opentracing.set_tag("device_keys", device_keys)
```

Comment:
don't we get this already from the `keys` arg tag? (likewise for one_time_keys below)

Reply:
True